File: code/31212.pas
program Bai212;
  var n, a, b,x, dem: integer;
  function TONGUOC(m: integer): integer;
    var i, s: integer;
  begin
  s := m;
  for i := 1 to m div 2 do 
     if m mod i = 0 then  s:= s + i;
 TONGUOC := s;
  end;
begin
  readln(a,b);
  dem := 0; 
  for x := a to b do 
          if TONGUOC(x) mod 2 = 1 then dem := dem + 1;
          write(dem);
end.

File: code/31212.cpp
#include <iostream>
using namespace std;

int tonguoc(int n){
    int s,x;
    s=0;
    for (x=1; x<=n; x++){
        if (n%x==0)
            s=s+x;
    }
    return s;
}

int main() 
{
    int a,b,dem,x;
    cin>>a>>b;
    dem=0;
    for (x=a; x<=b; x++)
        if (tonguoc(x)%2==1)
            dem++;
    cout<<dem;
    return 0;
}